junkie/context.py

```diff
@@ -19,18 +19,18 @@ def add(self, *singletons_and_factories_args: Union[Dict[str, object], Dict[str,
                     self._singletons[key] = value
 
     @contextmanager
-    def build(self, instances: Union[Set[str], List[str], type, str]):
-        if isinstance(instances, set) or isinstance(instances, list):
-            with self.build_dict(instances) as instance_dict:
+    def build(self, names_or_type: Union[Set[str], List[str], str, type]):
```

Comment:
`names_or_types`?

Reply:
Oder geht da nur ein type Argument?

Reply:
Indeed, currently you can only have a single type here. All sets and lists expect strings.
Do you think both sould be supported (str and type in lists and sets)?
PR? ;-)